actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -822,7 +822,7 @@ def url_for(options, route_name = nil, url_strategy = UNKNOWN, method_name = nil
         path = route_with_params.path(method_name)
         params = route_with_params.params
 
-        if options[:params].is_a?(Hash)
+        if options[:params].is_a?(Hash) || options[:params].is_a?(ActionController::Parameters)
```

Comment:
I was thinking that anything mergeable should fall in here. Instead of having an exhaustive list of mergeable types 🤔 
```suggestion
        if options[:params].respond_to? :to_h
```

Reply:
Maybe yeah. Would need to be `:to_hash` though, as `to_h` is explicit casting, and `to_hash` implicit.

Reply:
Ok, did just that. If that's good with you I'll merge on green.